io/teca_cf_writer.h

```diff
@@ -10,8 +10,48 @@
 
 TECA_SHARED_OBJECT_FORWARD_DECL(teca_cf_writer)
 
+
+// A writer for Cartesian meshes in NetCDF CF2 format.
 /**
-an algorithm that writes cartesian meshes in NetCDF CF format.
+Writes data to NetCDF CF2 format. This algorithm is conceptually an execution
+engine capable of driving the above pipeline with our without threads and
+stream results in the order that they are genreated placing them in the correct
+location in the output dataset. The output dataset is a collection of files
+each with a user specified number of time steps per file. The total number of
+time steps in the dataset is determined by the combination of the number of
+time steps in the input dataset and user defined subsetting if any. The writer
+uses MPI collective I/O to produce the files. In parallel time steps are mapped
+to ranks such that each rank has approximately the same number of time steps.
+Incoming steps are mapped to files. A given MPI rank may be writing to multiple
+files. The use of MPI collecitves implies care must be taken in its use to
+avoid deadlocks.
+
+Due to the use of MPI collectives I/O certain information must be known during
+the report phase of pipeline execution, before the execute phase of pipeline
+execution begins. The information that is needed is:
+
+number of time steps - specified by the pipeline control index_initializer key
+                       found in metadata produced by the source (eg CF reader)
+
+extent - 6 64 bit integers defining the 3 spatial dimensions of each time step
+         found in metadata produced by the source (eg CF reader)
+
+point arrays - list of strings naming the point centered arrays that will be
+               written. set by the user prior to execution by writer properties.
+
+infomration arrays - list of strings naming the non-geometric arrays that will
```

Comment:
how are the dimensions of the information arrays defined?  It seems obvious that the point arrays would inherit these from the input data, and the information arrays need to have their dimensions defined.  It might be useful to briefly comment on what part of the API is responsible for setting the dimension names/sizes/metadata for information array dimensions.

Reply:
added reference to size attribute